src/olympia/abuse/models.py

```diff
@@ -172,40 +173,65 @@ def get_cinder_reporter(cls, abuse):
             )
         return reporter
 
-    @classmethod
-    def handle_already_removed(cls, abuse_report):
-        entity_helper = cls.get_entity_helper(
-            abuse_report.target,
-            addon_version_string=abuse_report.addon_version,
-            resolved_in_reviewer_tools=False,
+    def should_auto_resolve(self):
+        target = self.target
+        is_disabled = (
+            getattr(target, 'deleted', False)
+            or getattr(target, 'banned', False)
+            or getattr(target, 'status', -1) == amo.STATUS_DISABLED
+        )
+        versions = (
```

Comment:
Evaluating this queryset is going to be potentially quite expensive on add-ons with a bunch of versions, and it's a bit wasteful if most were not reported.

I think you should wait until `report_qs` is built, extract `addon_version` for each report into a set and then build `versions` by filtering the queryset with that set.

Reply:
https://github.com/mozilla/addons-server/pull/23465/commits/fd8e5174ad43534b7d20710eda3be185bf5c05a9